pkg/admissioncontroller/webhook/admission/resourcesize/handler.go

```diff
@@ -155,14 +217,60 @@ func isUnrestrictedUser(userInfo authenticationv1.UserInfo, subjects []rbacv1.Su
 	return userMatch(userInfo, subjects)
 }
 
-func findLimitForGVR(limits []admissioncontrollerconfigv1alpha1.ResourceLimit, gvr *metav1.GroupVersionResource) *resource.Quantity {
+func findRestrictionsForGVR(limits []admissioncontrollerconfigv1alpha1.ResourceLimit, gvr *metav1.GroupVersionResource) (*resource.Quantity, *int64) {
 	for _, limit := range limits {
 		size := limit.Size
+		count := limit.Count
 		if admissioncontrollerhelper.APIGroupMatches(limit, gvr.Group) &&
 			admissioncontrollerhelper.VersionMatches(limit, gvr.Version) &&
 			admissioncontrollerhelper.ResourceMatches(limit, gvr.Resource) {
-			return &size
+			return size, count
 		}
 	}
-	return nil
+	return nil, nil
+}
+
+func (h *Handler) getKindFromGVR(gvr *metav1.GroupVersionResource) (string, error) {
+	if h.RESTMapper == nil {
+		return "", errors.New("RESTMapper not initialized")
+	}
+
+	gvk, err := h.RESTMapper.KindFor(schema.GroupVersionResource{
+		Group:    gvr.Group,
+		Version:  gvr.Version,
+		Resource: gvr.Resource,
+	})
+	if err != nil {
+		return "", err
+	}
+	return gvk.Kind, nil
+}
+
+func (h *Handler) countExistingResources(_ admission.Request, gvr *metav1.GroupVersionResource) (int64, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+
+	// Get the proper Kind for this resource
+	kind, err := h.getKindFromGVR(gvr)
+	if err != nil {
+		h.Logger.Error(err, "Failed to determine kind from GVR", "gvr", gvr)
+		return 0, nil
+	}
+
+	list := &metav1.PartialObjectMetadataList{}
+	list.SetGroupVersionKind(schema.GroupVersionKind{
+		Group:   gvr.Group,
+		Version: gvr.Version,
+		Kind:    kind + "List",
+	})
+
+	// List all resources of this type cluster-wide (since we only count non-namespaced resources)
+	if err := h.Client.List(ctx, list); err != nil {
```

Comment:
```suggestion
	if err := h.Client.List(ctx, list, client.Limit(count + 1)); err != nil {
```

You could limit the count of objects retrieved here.